### ISSUE TITLE
Typos and inconsistent naming in trace translators.

### ISSUE THREAD

**Issue by SamWitty** [probcomp/Gen.jl#385]
When running the example code in https://www.gen.dev/dev/ref/trace_translators/#General-Trace-Translators-1, julia throws the following error:

`ERROR: UndefVarError: proposal not defined
Stacktrace:
 [1] (::GeneralTraceTranslator)(::Gen.DynamicDSLTrace{DynamicDSLFunction{Any}}; check::Bool, prev_observations::EmptyChoiceMap) at /Users/switty/.julia/packages/Gen/unzLD/src/inference/trace_translators.jl:727
 [2] (::GeneralTraceTranslator)(::Gen.DynamicDSLTrace{DynamicDSLFunction{Any}}) at /Users/switty/.julia/packages/Gen/unzLD/src/inference/trace_translators.jl:727
 [3] top-level scope at none:1`

I think the issue will be fixed by replacing `proposal` with `translator.q_forward` at https://github.com/probcomp/Gen.jl/blob/a96a77991e0e43f208272e9241c8f2434ffdedbf/src/inference/trace_translators.jl#L727, but I don't know enough about the codebase to feel comfortable making that change.

For completeness, here is the code that generates the error.

```
@gen function p1()
    x ~ uniform(0, 1)
    y ~ uniform(0, 1)
end

@gen function p2()
    i ~ uniform_discrete(1, 10) # interval [(i-1)/10, i/10]
    j ~ uniform_discrete(1, 10) # interval [(j-1)/10, j/10]
end

@gen function q1()
end

@gen function q2(p2_trace)
    i = p2_trace[:i]
    j = p2_trace[:j]
    dx ~ uniform(0.0, 0.1)
    dy ~ uniform(0.0, 0.1)
end

@transform f (p1_trace, q1_trace) to (p2_trace, q2_trace) begin
    x = @read(p1_trace[:x], :continuous)
    y = @read(p1_trace[:y], :continuous)
    i = ceil(x * 10)
    j = ceil(y * 10)
    @write(p2_trace[:i], i, :discrete)
    @write(p2_trace[:j], j, :discrete)
    @write(q2_trace[:dx], x / 10, :continuous)
    @write(q2_trace[:dy], y / 10, :continuous)
end

@transform f_inv (p2_trace, q2_trace) to (p1_trace, q1_trace) begin
    i = @read(p2_trace[:i], :discrete)
    j = @read(p2_trace[:j], :discrete)
    dx = @read(q2_trace[:dx], :continuous)
    dy = @read(q2_trace[:dy], :continuous)
    x = (i-1)/10 + dx
    y = (j-1)/10 + dy
    @write(p1_trace[:x], x, :continuous)
    @write(p1_trace[:y], x, :continuous)
end

pair_bijections!(f, f_inv)

translator = GeneralTraceTranslator(
    p_new=p2,
    p_new_args=(),
    new_observations=choicemap(),
    q_forward=q1,
    q_forward_args=(),
    q_backward=q2,
    q_backward_args=(),
    f=f)

t1 = simulate(p1, ())

(t2, log_weight) = translator(t1)
```

**Comment by ztangent** [comment]
Adding another a typo I noticed to this issue: `q_backard::GenerativeFunction` at line 712.

https://github.com/probcomp/Gen.jl/blob/a96a77991e0e43f208272e9241c8f2434ffdedbf/src/inference/trace_translators.jl#L708-L721

**Comment by ztangent** [comment]
Figured I'd also add a few other odd things I noticed (I apologies for not catching these since I was supposed to review the original PR):

1. In `DeterministicTraceTranslator`, there is a call to `check_observations` call that I believe will result in a crash. Should `observations` instead be `translator.new_observations`?

https://github.com/probcomp/Gen.jl/blob/a96a77991e0e43f208272e9241c8f2434ffdedbf/src/inference/trace_translators.jl#L661-L667

2. In `SimpleExtendingTraceTranslator`, the field names for new observations are inconsistent with all other translators: `new_obs` instead of `new_observations`, `q_fwd` instead of `q_forward`.

3. As a minor coding style suggestion, perhaps all functions of the form `symmetric_trace_translator_run_transform` should just be replaced with `run_transform` functions which dispatch based on the type of the translator (given as the first argument).
